sdk/src/main/java/com/mapbox/maps/renderer/MapboxWidgetRenderer.kt

```diff
@@ -0,0 +1,171 @@
+package com.mapbox.maps.renderer
+
+import android.opengl.GLES20
+import com.mapbox.common.Logger
+import com.mapbox.maps.renderer.egl.EGLCore
+import com.mapbox.maps.renderer.widget.Widget
+import java.util.concurrent.CopyOnWriteArraySet
+import javax.microedition.khronos.egl.EGLContext
+import javax.microedition.khronos.egl.EGLSurface
+
+internal class MapboxWidgetRenderer(
+  private val translucentSurface: Boolean,
+  private val antialiasingSampleCount: Int,
+) {
+  private var widgetEglPrepared = false
+  private lateinit var widgetEglSurface: EGLSurface
+  private lateinit var widgetEglCore: EGLCore
+
+  private val textures = intArrayOf(0)
+  private val framebuffers = intArrayOf(0)
+
+  private val widgets = CopyOnWriteArraySet<Widget>()
+
+  private var width = 0
+  private var height = 0
+
+  fun getTextureId() = textures[0]
+
+  val needRender: Boolean
+    get() = widgets.any { it.renderer.needRender }
+
+  fun onSharedContext(sharedContext: EGLContext) {
+    if (widgetEglPrepared) {
+      TODO("New shared context while previous is still alive!")
```

Comment:
replace TODO with at least log I guess or simply recreate EGL with new shared context. I believe that situation may be real, did you test it? we should have new context e.g. when we minimize / maximize an app

Reply:
Done